File: FEB20/03_FEB_2020/03_FEB_2020.py
# Author- Lakhan Nad
s = input()
countarr = [0]*26;
i = 0
while i < len(s):
    x = ord(s[i])-ord('a')
    c = 1;
    j = 1;
    while(i+j < len(s) and s[i+j] == s[i]):
        c += 1
        j += 1
    i = i + j
    if(countarr[x] < c):
        countarr[x] = c
queries_count = int(input())
for _ in range(queries_count):
    flag = False
    z = int(input())
    for y in range(1,27):
        if(z%y==0 and z//y <= countarr[y-1]):
            flag = True
    if(flag):
        print("Yes")
    else:
        print("No")    

File: FEB20/03_FEB_2020/03_FEB_2020.cpp
// Author- Lakhan Nad
#include <bits/stdc++.h>

using namespace std;

vector<string> weightedUniformStrings(string s, vector<int> queries) {
  vector<string> result;
  int arr[26] = {0};
  char c;
  int count;
  for (int i = 0; i < s.length(); i++) {
    c = s[i];
    count = 0;
    while (i < s.length() && s[i] == c) {
      count++;
      i++;
    }
    if (arr[c - 'a'] < count) {
      arr[c - 'a'] = count;
    }
    i--;
  }
  bool present;
  for (int i = 0; i < queries.size(); i++) {
    present = false;
    for (int j = 1; j < 27; j++) {
      if (queries[i] % j == 0 && arr[j - 1] * j >= queries[i]) {
        present = true;
        break;
      }
    }
    if (present) {
      result.push_back("Yes");
    } else {
      result.push_back("No");
    }
  }
  return result;
}

int main() {
  ofstream fout(getenv("OUTPUT_PATH"));
  string s;
  getline(cin, s);
  int queries_count;
  cin >> queries_count;
  cin.ignore(numeric_limits<streamsize>::max(), '\n');
  vector<int> queries(queries_count);
  for (int i = 0; i < queries_count; i++) {
    int queries_item;
    cin >> queries_item;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');
    queries[i] = queries_item;
  }
  vector<string> result = weightedUniformStrings(s, queries);
  for (int i = 0; i < result.size(); i++) {
    fout << result[i];

    if (i != result.size() - 1) {
      fout << "\n";
    }
  }
  fout << "\n";
  fout.close();
  return 0;
}